crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -84,29 +89,26 @@ fn site_packages_dir_from_sys_prefix(
         // the `pyvenv.cfg` file anyway, in which case we could switch to that method
         // rather than iterating through the whole directory until we find
         // an entry where the last component of the path starts with `python3.`
-        if !path
-            .components()
-            .next_back()
-            .is_some_and(|last_part| last_part.as_str().starts_with("python3."))
```

Comment:
I think this was incorrect because it didn't continue if `next_back` returns `None`